web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+ */
+
+/**
+ * @typedef SpacePolicy
+ * @type {object}
+ * @property {string} name
+ * @property {string} label
+ * @property {string} description
+ */
+
+/** @type {SpacePolicy[]} */
+const SPACE_POLICIES = [
+  {
+    name: "delete",
+    label: N_("Delete current content"),
+    description: N_("All partitions will be removed and any data in the disks will be lost.")
+  },
+  {
+    name: "resize",
+    label: N_("Shrink existing partitions"),
+    description: N_("The data is kept, but the current partitions will be resized as needed.")
+  },
+  {
+    name: "keep",
+    label: N_("Use available space"),
+    description: N_("Existing partitions and data will not be modified. Only the space not assigned to any partition will be used.")
+  },
+  {
+    name: "custom",
+    label: N_("Custom"),
+    description: N_("Select what to do with each partition.")
+  }
+];
+
+// Names of the columns for the policy actions.
+const columnNames = {
+  device: N_("Used device"),
+  content: N_("Current content"),
+  size: N_("Size"),
+  action: N_("Action")
+};
+
+/**
+ * Indicates whether a device is a drive (disk, RAID).
+ * @function
+ *
+ * @param {StorageDevice} device
+ * @returns {boolean}
+ */
+const isDrive = (device) => Object.keys(device).includes("vendor");
+
+/**
+ * Column content with the description of a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceDescriptionColumn = ({ device }) => {
+  return (
+    <>
+      <div>{device.name}</div>
+      <If
+        condition={isDrive(device)}
+        then={<div className="fs-small">{`${device.vendor} ${device.model}`}</div>}
+      />
+    </>
+  );
+};
+
+/**
+ * Column content with information about the current content of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceContentColumn = ({ device }) => {
+  const PartitionTableContent = () => {
+    const numPartitions = device.partitionTable.partitions.length;
+
+    return (
+      <>
+        <div>
+          {sprintf(n_("%d partition", "%d partitions", numPartitions), numPartitions)}
+        </div>
+        <div className="fs-small">
+          {sprintf(_("%s partition table"), device.partitionTable.type.toUpperCase())}
+        </div>
+      </>
+    );
+  };
+
+  const BlockContent = () => {
+    const systems = device.systems;
+    const filesystem = device.filesystem;
+
+    const content = () => {
+      if (systems.length > 0) return systems.join(", ");
+      if (device.filesystem?.isEFI) return _("EFI");
+
+      return _("Not identified");
+    };
+
+    return (
+      <>
+        <div>{content()}</div>
+        <If
+          condition={filesystem}
+          then={
+            <div className="fs-small">
+              {sprintf(_("%s file system"), filesystem?.type)}
+            </div>
+          }
+        />
+      </>
+    );
```

Comment:
NP: I do not know exactly why, but I do not like this kind  of methods. I prefer a `Component` instead.

```suggestion
    const Content = () => {
      if (systems.length > 0) return systems.join(", ");
      if (device.filesystem?.isEFI) return _("EFI");

      return _("Not identified");
    };

    return (
      <>
        <div><Content />}</div>
        <If
          condition={filesystem}
          then={
            <div className="fs-small">
              {sprintf(_("%s file system"), filesystem?.type)}
            </div>
          }
        />
      </>
    );
```

But I'm aware that we're kind of _abusing_ of components here. Or maybe not.

Another option is to prefix the function with the `render` word. I.e., `renderContent` or `renderWhatever`. It makes the code more intuitive, IMHO.

Also wondering if this kind of "<DeviceContent />"  internal component could make sense in another place of the UI. E.g., in the DeviceSelector and also in another table. If so, it looks like a helper component to me, and consequently we should put it in a more central/common place to avoid repeating almost the same in multiple places.